cloudify_vsphere/resource_pool/__init__.py

```diff
@@ -22,47 +22,149 @@
+        spec.memoryAllocation = vim.ResourceAllocationInfo()
+        spec.memoryAllocation.limit = -1
+        spec.memoryAllocation.reservation = 0
+        spec.memoryAllocation.expandableReservation = True
+        spec.memoryAllocation.shares = vim.SharesInfo()
+        spec.memoryAllocation.shares.shares = 163840
+        spec.memoryAllocation.shares.level = 'normal'
+        spec.cpuAllocation = vim.ResourceAllocationInfo()
+        spec.cpuAllocation.limit = -1
+        spec.cpuAllocation.reservation = 0
+        spec.cpuAllocation.expandableReservation = True
+        spec.cpuAllocation.shares = vim.SharesInfo()
+        spec.cpuAllocation.shares.shares = 4000
+        spec.cpuAllocation.shares.level = 'normal'
+
+    # override values given the input
+    if 'memoryAllocation' in pool_spec:
+        if 'limit' in pool_spec.get('memoryAllocation'):
+            spec.memoryAllocation.limit = \
+                pool_spec.get('memoryAllocation').get('limit')
+        if 'reservation' in pool_spec.get('memoryAllocation'):
+            spec.memoryAllocation.reservation = \
+                pool_spec.get('memoryAllocation').get('reservation')
+        if 'expandableReservation' in pool_spec.get('memoryAllocation'):
+            spec.memoryAllocation.expandableReservation = \
+                pool_spec.get('memoryAllocation').get('expandableReservation')
+        if 'shares' in pool_spec.get('memoryAllocation'):
+            shares = pool_spec.get('memoryAllocation').get('shares')
+            if 'shares' in shares:
+                spec.memoryAllocation.shares.shares = shares.get('shares')
+            if 'level' in shares:
+                spec.memoryAllocation.shares.level = shares.get('level')
+
+    if 'cpuAllocation' in pool_spec:
+        if 'limit' in pool_spec.get('cpuAllocation'):
+            spec.cpuAllocation.limit = \
+                pool_spec.get('cpuAllocation').get('limit')
+        if 'reservation' in pool_spec.get('cpuAllocation'):
+            spec.cpuAllocation.reservation = \
+                pool_spec.get('cpuAllocation').get('reservation')
+        if 'expandableReservation' in pool_spec.get('cpuAllocation'):
+            spec.cpuAllocation.expandableReservation = \
+                pool_spec.get('cpuAllocation').get('expandableReservation')
+        if 'shares' in pool_spec.get('cpuAllocation'):
+            shares = pool_spec.get('cpuAllocation').get('shares')
+            if 'shares' in shares:
+                spec.cpuAllocation.shares.shares = shares.get('shares')
+            if 'level' in shares:
+                spec.cpuAllocation.shares.level = shares.get('level')
+
+    return spec
+
 
 @op
 @with_server_client
-def create(ctx, server_client, name, use_external_resource):
-    vmware_resource = server_client._get_obj_by_name(
-        vim.ResourcePool,
-        name,
-    )
-
+def create(ctx, server_client, name, use_external_resource, host_name,
+           cluster_name, pool_spec):
```

Comment:
Add default values to new parameters, e.g. `host_name=None, cluster_name=None, pool_spec=None`